sqlite/ext/comdb2/tunables.c

```diff
@@ -0,0 +1,207 @@
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#if (!defined(SQLITE_CORE) || defined(SQLITE_BUILDING_FOR_COMDB2)) &&          \
+    !defined(SQLITE_OMIT_VIRTUALTABLE)
+
+#if defined(SQLITE_BUILDING_FOR_COMDB2) && !defined(SQLITE_CORE)
+#define SQLITE_CORE 1
+#endif
+
+#include <assert.h>
+#include "comdb2systbl.h"
+#include "tunables.h"
+#include "comdb2.h"
+
+/*
+  comdb2_tunables: query various attributes of tunables.
+
+  TODO(Nirbhay): Check user permissions
+*/
+
+typedef struct {
+    sqlite3_vtab_cursor base; /* Base class - must be first */
+    sqlite3_int64 rowid;      /* Row ID */
+} systbl_tunables_cursor;
+
+/* Column numbers (always keep the below table definition in sync). */
+enum {
+    TUNABLES_COLUMN_NAME,
+    TUNABLES_COLUMN_DESCR,
+    TUNABLES_COLUMN_TYPE,
+    TUNABLES_COLUMN_VALUE,
+    TUNABLES_COLUMN_READONLY,
+};
+
+static int systblTunablesConnect(sqlite3 *db, void *pAux, int argc,
+                                 const char *const *argv, sqlite3_vtab **ppVtab,
+                                 char **pErr)
+{
+    int rc;
+
+    rc = sqlite3_declare_vtab(db, "CREATE TABLE comdb2_tunables(\"name\", "
+                                  "\"description\", \"type\", \"value\", "
+                                  "\"read_only\")");
+
+    if (rc == SQLITE_OK) {
+        if ((*ppVtab = sqlite3_malloc(sizeof(sqlite3_vtab))) == 0) {
+            return SQLITE_NOMEM;
+        }
+        memset(*ppVtab, 0, sizeof(*ppVtab));
+    }
+
+    return 0;
+}
+
+static int systblTunablesBestIndex(sqlite3_vtab *tab,
+                                   sqlite3_index_info *pIdxInfo)
+{
+    return SQLITE_OK;
+}
+
+static int systblTunablesDisconnect(sqlite3_vtab *pVtab)
+{
+    sqlite3_free(pVtab);
+    return SQLITE_OK;
+}
+
+static int systblTunablesOpen(sqlite3_vtab *p, sqlite3_vtab_cursor **ppCursor)
+{
+    systbl_tunables_cursor *cur =
+        sqlite3_malloc(sizeof(systbl_tunables_cursor));
+    if (cur == 0) {
+        return SQLITE_NOMEM;
+    }
+    memset(cur, 0, sizeof(*cur));
+    *ppCursor = &cur->base;
+    return SQLITE_OK;
+}
+
+static int systblTunablesClose(sqlite3_vtab_cursor *cur)
+{
+    sqlite3_free(cur);
+    return SQLITE_OK;
+}
+
+static int systblTunablesFilter(sqlite3_vtab_cursor *pVtabCursor, int idxNum,
+                                const char *idxStr, int argc,
+                                sqlite3_value **argv)
+{
+    systbl_tunables_cursor *pCur = (systbl_tunables_cursor *)pVtabCursor;
+    pCur->rowid = 0;
+    return SQLITE_OK;
+}
+
+static int systblTunablesNext(sqlite3_vtab_cursor *cur)
+{
+    systbl_tunables_cursor *pCur = (systbl_tunables_cursor *)cur;
+    pCur->rowid++;
```

Comment:
That won't be needed as vtable's eof  function (xEof) is called after every next to check if we have reached the end.